src/python/grpcio_tests/tests/unit/_metadata_code_details_test.py

```diff
@@ -363,6 +380,102 @@ def testCustomCodeStreamStream(self):
         self.assertIs(_NON_OK_CODE, exception_context.exception.code())
         self.assertEqual(_DETAILS, exception_context.exception.details())
 
+    def testAbortedUnaryUnary(self):
+        self._servicer.set_code(_NON_OK_CODE)
+        self._servicer.set_details(_DETAILS)
+        self._servicer.set_call_abort()
+
+        with self.assertRaises(grpc.RpcError) as exception_context:
+            self._unary_unary.with_call(object(), metadata=_CLIENT_METADATA)
+
+        self.assertTrue(
+            test_common.metadata_transmitted(
+                _CLIENT_METADATA, self._servicer.received_client_metadata()))
+        self.assertTrue(
+            test_common.metadata_transmitted(
+                _SERVER_INITIAL_METADATA,
+                exception_context.exception.initial_metadata()))
+        self.assertTrue(
+            test_common.metadata_transmitted(
+                _SERVER_TRAILING_METADATA,
+                exception_context.exception.trailing_metadata()))
+        self.assertIs(_NON_OK_CODE, exception_context.exception.code())
+        self.assertEqual(_DETAILS, exception_context.exception.details())
+
+    def testAbortedUnaryStream(self):
+        self._servicer.set_code(_NON_OK_CODE)
+        self._servicer.set_details(_DETAILS)
+        self._servicer.set_call_abort()
+
+        call = self._unary_stream(
+            _SERIALIZED_REQUEST, metadata=_CLIENT_METADATA)
+        received_initial_metadata = call.initial_metadata()
+        with self.assertRaises(grpc.RpcError):
+            for _ in call:
```

Comment:
`next(call)`? Or if you need to exhaust the entire iterator, `list(call)`?

Reply:
I noticed this, but after I saw the tests passed. So I wonder if that issue is fixed or do we not have a test to confirm? I am leaning to keep it as is to see the failure if we end up failing in the future?

Reply:
Now that I reread this old code: because the test calls `self._servicer.set_call_abort()` before even invoking the RPC, the response iterator will raise with the very first attempt to read a message. So in this circumstance `next(call)` is guaranteed to raise an exception, and we needn't have the test exhaust the response iterator.

Correct me if I'm introducing a race condition and flaky test? No one likes flaky tests.